src/Tree/RowMenu.tsx

```diff
@@ -0,0 +1,110 @@
+import { KeyboardArrowRight } from '@mui/icons-material';
+import {
+  Button,
+  CircularProgress,
+  IconButton,
+  Stack,
+  Typography,
+  styled,
+} from '@mui/material';
+
+import { useState } from 'react';
+
+import { ItemType } from '@graasp/sdk';
+
+import { ItemIcon } from '..';
+import type { NavigationElement } from './types';
+
+export interface RowMenuProps {
+  item: NavigationElement;
+  onNavigate: (item: NavigationElement) => void;
+  selectedId?: string;
+  onClick: (item: NavigationElement) => void;
+  isDisabled?: (item: NavigationElement) => boolean;
+  id?: string;
+  arrowId?: string;
+}
+
+const StyledButton = styled(Button, {
+  shouldForwardProp: (prop) => prop !== 'isSelected',
+})<{ isSelected: boolean }>(({ theme, isSelected }) => ({
+  color: theme.palette.text.primary,
+  width: '100%',
+  justifyContent: 'start',
+  background: isSelected ? theme.palette.grey[200] : 'none',
+  textTransform: 'none',
+  pl: 1,
+}));
+
+const RowMenu = ({
+  item,
+  onNavigate,
+  onClick,
+  selectedId,
+  isDisabled,
+  id,
+  arrowId,
+}: RowMenuProps): JSX.Element | null => {
+  const [isHoverActive, setIsHoverActive] = useState(false);
```

Comment:
I think a state is not needed. See this demo, you can add a bit of css to toggle the visibility of the IconButton when the parent is hovered. I should be more responsive (less latency) and reduce code and logic.

https://github.com/graasp/graasp-ui/assets/39373170/c7bb0359-64b6-4ef7-bb38-79cbbf6950e0